Ex1/ex1_student_solution.py

```diff
@@ -127,28 +128,28 @@ def compute_forward_homography_fast(
         # return new_image
         """INSERT YOUR CODE HERE"""
         dst_image = np.zeros(dst_image_shape, dtype=src_image.dtype)
-        h1, w1 = src_image.shape[:-1]
-        h2, w2 = dst_image.shape[:-1]
+        h_s, w_s = src_image.shape[:-1]
+        h_d, w_d = dst_image.shape[:-1]
 
         # 1
-        U, V = np.meshgrid(np.arange(w1), np.arange(h1))
-        u = U.reshape(1, -1)
-        v = V.reshape(1, -1)
-        z = np.ones((1, h1 * w1))
+        U, V = np.meshgrid(np.arange(w_s), np.arange(h_s))
+        u_s = U.reshape(1, -1)
+        v_s = V.reshape(1, -1)
+        z_s = np.ones((1, h_s * w_s))
         # 2
-        P = np.concatenate((u, v, z))
+        pts_s = np.concatenate((u_s, v_s, z_s))
         # 3
-        P_tag = np.dot(homography, P)
+        pts_d = homography @ pts_s
         # 4
-        P_tag = P_tag / P_tag[-1, :]
-        U_tag, V_tag = np.round(P_tag[0:2]).astype(np.int)
-        valid_coords = np.where((U_tag >= 0) & (U_tag < w2) & (V_tag >= 0) & (V_tag < h2))[0]
-        u = u.reshape(-1)[valid_coords]
-        v = v.reshape(-1)[valid_coords]
-        u_tag = U_tag[valid_coords]
-        v_tag = V_tag[valid_coords]
+        pts_d /= pts_d[-1, :]
+        u_d, v_d = np.round(pts_d[0:2]).astype(np.int)
+        valid_coords = np.where((u_d >= 0) & (u_d < w_d) & (v_d >= 0) & (v_d < h_d))[0]
+        u_s = u_s.reshape(-1)[valid_coords]
```

Comment:
I know that I though that it is good to keep the same variable name but maybe it will be clearer if we change to smth like u_s_in_dst_frame...  (if you think it is clear enough, don't change)